src/VisualStudio/Core/Def/Implementation/ProjectSystem/CPS/IWorkspaceProjectContext.cs

```diff
@@ -58,5 +60,26 @@ internal interface IWorkspaceProjectContext : IDisposable
         void EndBatch();
 
         void ReorderSourceFiles(IEnumerable<string> filePaths);
+
+        /// <summary>
+        /// Applies changes to symbols for documents after their file path has changed. The same policy
+        /// is applied across all documents based on the <see cref="FilePathChangeSymbolChange"/>
+        /// </summary>
+        /// <param name="documentIdToOriginalPathMapping">Mapping of document ids to the original file path. Original path may be used to determine if symbol changes are applicable</param>
+        /// <remarks>
+        /// Note that calls to this may do expensive work to fix up symbol renaming. Symbol changes express a desire to change the symbol, but may not result in a symbol change if 
+        /// deemed unnecessary. For example, if no types matched the file name before then a named type change will not be performed. 
+        /// </remarks>
+        void ApplyDocumentSymbolChanges(ImmutableDictionary<DocumentId, string> documentIdToOriginalPathMapping, FilePathChangeSymbolChange filePathChangeSymbolChange, CancellationToken cancellationToken);
     }
+
+    [Flags]
+    internal enum FilePathChangeSymbolChange
```

Comment:
Is this something the project system wants to control? If we add a new one, are they going to want to have to update this?

Reply:
I think yes, only because there is some user configuration that is applicable here. Although None doesn't really seem valid so I'll remove that... 

For example, the current UI only supports named type renaming. We haven't fully flushed out the design for namespace as well, but I imagine it should be configurable to turn off. We want this to feel like magic, and we assume users want it, but they should be able to disable if they find it disruptive to their workflow.

Reply:
But if it was configurable, it's not the project system's job to decide that configuration. We'd be the ones figuring it out.

Reply:
I might be conflating what is going on today with what we want. I see your point, we could start reading configuration as needed with this API. Add a VS config provider for this setting, and have a default that is... all of them? none of them?

Reply:
Changed design to get rid of this. Roslyn will handle policy on this.